File: 0011-container-with-most-water/0011-container-with-most-water.js
var maxArea = function (height) {
  let left = 0;
  let right = height.length - 1;
  let maxVol = Math.min(height[left], height[right]) * (right - left);
  while (left < right) {
    let curr = Math.min(height[left], height[right]) * (right - left);
    maxVol = Math.max(curr, maxVol);
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }
  return maxVol;
};

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution
{
public:
  int maxArea(vector<int> &height)
  {
    int left = 0;
    int right = height.size() - 1;
    int maxVol = min(height[left], height[right]) * (right - left);
    while (left < right)
    {
      int vol = min(height[left], height[right]) * (right - left);
      maxVol = max(vol, maxVol);
      if (height[left] < height[right])
      {
        left++;
      }
      else
      {
        right--;
      }
    }
    return maxVol;
  }
};